plugins/InspectionGadgets/src/com/siyeh/ig/bugs/IgnoreResultOfCallInspection.java

```diff
@@ -267,6 +267,9 @@ else if (call instanceof PsiMethodReferenceExpression) {
         exception = KNOWN_EXCEPTIONAL_SIDE_EFFECTS.mapFirst((PsiMethodReferenceExpression)call);
       }
       if (exception == null) return false;
+      if (isInTestContainer(call)) {
```

Comment:
This does not seem to be the correct place to check this. If the inspection setting `report all ignored non-library call` is enabled non-library calls are still reported in a test container. This does not seem to be desired.

Reply:
Thank you for this review!
Honestly, it is a really interesting point, because before your comment I thought that IDEA should report this case anyway. Because a case like 
```java
        class X {
          private static String twice(String s) {
            return s+s;
          }
          public void test(String abc){
            org.junit.jupiter.api.Assertions.assertThrows(RuntimeException.class, ()-> {
                twice(abc);
            });
          }
        }
```
would seem to be an error.

After reading your comment, I discovered several cases in which it would be useful. So, additionally, I added one more option 
`Don't consider non-library calls inside methods of AssertJ or JUnit as ignored` to process these cases. 

Thank you!